src/Program.cs

```diff
@@ -1,245 +1,263 @@
-                                             type = "Microsoft.Search/searchServices",
-                                             location = "West US",
-                                             properties = new
-                                             {
-                                                 sku = new { name = "standard" },
-                                                 partitionCount = 1,
-                                                 replicaCount = 2
-                                             }
-                                         });
-            DumpResponse("Dynamically scale Azure Search service", response);
+            foreach (var group in groups)
+            {
+                var resources = await resourceClient.Resources.ListAsync("resourceGroup eq '" + group.Name + "' and resourceType eq 'Microsoft.Search/searchServices'");
+                if (resources.Count() > 0)
+                    Console.WriteLine("resourceGroup: {0}", group.Name);
+                {
+                    foreach (var resource in resources)
+                    {
+                        Console.WriteLine("   service name: {0}, sku: {1}, location: {2}", resource.Name, resource.Sku.Name, resource.Location);
+                    }
+                }
+            }
+            Console.WriteLine();
 
-            // Delete search service
-            response = ExecuteArmRequest(HttpMethod.Delete, "resourcegroups/Default-Web-WestUS/providers/Microsoft.Search/searchServices/" + name + "?api-version=2014-07-31-Preview");
-            DumpResponse("Delete Azure Search service", response);
-        }
+            // Create a new search service called "sampleservice#" (where # is a random number, to make it less likely to have collisions)
+            string newServiceName = "sampleservice" + _random.Next(0, 1000000).ToString();
+            string newGroupName = "samplegroup" + _random.Next(0, 1000000).ToString();
+            await resourceClient.ResourceGroups.CreateOrUpdateAsync(newGroupName, new ResourceGroup { Location = "West US" });
+
+            // PLEASE NOTE: By default, the code below will create a Standard (S1) search service which is a paid service tier.
+            // You can change this to the free service tier, but if you already have a free service in this subscription the operation will fail.
+            var newService = await searchClient.Services.CreateOrUpdateAsync(newGroupName, newServiceName,
+                new SearchService() {
+                    Location = "West US",
+                    Sku = new Sku() { Name = SkuName.Standard }, // use "standard" for standard services
+                    PartitionCount = 1,
+                    ReplicaCount = 1
+                });
+
+            // Wait for service provisioning to complete
+            while (newService.ProvisioningState == ProvisioningState.Provisioning)
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(10));
```

Comment:
In an async method, you should use `Task.Delay`, not `Thread.Sleep`. It doesn't matter so much for a console app, but it would hurt scalability in a web app and we want to set a good example.

Reply:
Done.  I'm curious if this is good async pattern in general.  Is there a better way to handle checking for the finish of a long running process like this in the SDK other than pinging the status in a loop?

Reply:
No, we don't support the more modern pattern of using HTTP 202 for long running operations. Polling is the best you can do.